src/app/shared/categories/categories.component.ts

```diff
@@ -1,44 +1,44 @@
-import { Component, OnDestroy } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
 import {
   categoriesList,
   PageTitles,
   ProductTypesType,
 } from '../constants/contstans';
-import { ICategories } from '../interfaces/interfaces';
-import { Store } from '@ngrx/store';
-import { Subscription } from 'rxjs';
-import {changeTypeAction} from "../../store/product-type-store/product-type.actions";
-
+import { AppStateInterface, ICategories } from '../interfaces/interfaces';
+import { select, Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { changeTypeAction } from '../../store/product-type-store/product-type.actions';
+import { productTypeSelector } from '../../store/product-type-store/product-type.selectors';
+import {
+  addProductsAction,
+  runLoadingAction,
+} from '../../store/products-store/products.actions';
 
 @Component({
   selector: 'app-categories',
   templateUrl: './categories.components.html',
 })
-export class CategoriesComponent implements OnDestroy {
+export class CategoriesComponent implements OnInit {
   title?: string;
   categoriesList?: ICategories[];
-  productTypeSubscription?: Subscription;
+  productTypeSubscription$?: Observable<string>;
```

Comment:
Seems like wrong name

Subscription and Observable it's 2 different abstraction. 
Seems like it should be just productType